crates/uv/src/commands/build_frontend.rs

```diff
@@ -1155,39 +1167,55 @@ enum BuildMessage {
 }
 
 impl BuildMessage {
-    /// The filename of the wheel or source distribution.
-    fn filename(&self) -> &DistFilename {
+    /// The normalized filename of the wheel or source distribution.
+    fn normalized_filename(&self) -> &DistFilename {
         match self {
-            BuildMessage::Build { filename: name, .. } => name,
-            BuildMessage::List { filename: name, .. } => name,
+            BuildMessage::Build {
+                normalized_filename: name,
+                ..
+            } => name,
+            BuildMessage::List {
+                normalized_filename: name,
+                ..
+            } => name,
+        }
+    }
+
+    /// The filename of the wheel or source distribution before normalization.
+    fn raw_filename(&self) -> &String {
```

Comment:
I'd suggest returning `&str` here? I think it's somewhat uncommon to return `&String` since it's strictly less flexible (you can always get an `&str` from an `&String`, but you can't go the other way around).